iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -29,22 +29,29 @@ namespace iox
 {
 namespace runtime
 {
-std::function<PoshRuntime&(const std::string& name)> PoshRuntime::s_runtimeFactory = PoshRuntime::defaultRuntimeFactory;
+std::function<PoshRuntime&(const ProcessName_t& name)> PoshRuntime::s_runtimeFactory =
+    PoshRuntime::defaultRuntimeFactory;
 
 
-PoshRuntime& PoshRuntime::defaultRuntimeFactory(const std::string& name) noexcept
+PoshRuntime& PoshRuntime::defaultRuntimeFactory(const ProcessName_t& name) noexcept
 {
     static PoshRuntime instance(name);
     return instance;
 }
 
 // singleton access
-PoshRuntime& PoshRuntime::getInstance(const std::string& name) noexcept
+PoshRuntime& PoshRuntime::getInstance(const ProcessName_t& name) noexcept
 {
     return PoshRuntime::s_runtimeFactory(name);
 }
 
-PoshRuntime::PoshRuntime(const std::string& name, const bool doMapSharedMemoryIntoThread) noexcept
+ProcessName_t& PoshRuntime::defaultRuntimeInstanceName() noexcept
+{
+    static ProcessName_t defaultInstanceName = "dummy";
```

Comment:
Again magic name? I think we should move the default name into `posh_types` like
`static constexpr char DEFAULT_INSTANCE_NAME[MAX_PROCESS_NAME_LENGTH] = "dummy";` and use it here.

Reply:
I deliberately changed the constexpr to this, because it's used as default value for the instance name in `getInstance(..)`. If we keep the constecpr, every call to `getInstance()` will construct a cxx::string

Reply:
In support of @elBoberido : we also do not want or need this name to be publicly accessible, which it would be in posh_types.I would keep it hidden, it is not really intended to construct a dummy runtime anyway.

Reply:
btw. I never liked the `getInstance()` call with a default parameter which is just a random string to indicate that the function was called without a parameter. I would suggest to have two functions `initRuntim(ProcessName_t& name)` and `getInstance()` with
```
private:
    PoshRuntime& getInstance(cxx::optional<ProcessName_t*> name);
public:
    PoshRuntime& initRuntime(ProcessName_t& name) {
        return getInstance(cxx::make_optional<ProcessName_t*>(&name));
    }
    PoshRuntime& getInstance() {
        return getInstance(cxx::nullopt);
    }
```
With this approach we get rid of a magic string and make the code more robust.
 Maybe @FerdinandSpitzschnueffler can implement this with #327 
@budrus, @elfenpiff @MatthiasKillat what do you think?

Reply:
Sounds good 👍

Reply:
Great idea! I think @FerdinandSpitzschnueffler had the same idea for the runtime factory.

Reply:
Good idea and quite similar to the runtime factory changes I implemented for [#327](https://github.com/eclipse/iceoryx/issues/327). I will add @elBoberido's approach.

Reply:
@FerdinandSpitzschnueffler since this will need changes is several places and make the PR of the bugfix quite large, maybe it would be better to have a separate PR for this

Reply:
will be done by #382